### DISCUSSION TITLE
posts/keepalived-ha-loadbalancer/

### DISCUSSION THREAD

**Discussion by giscus** [techno-tim/techno-tim.github.io#274] [Comments]
# posts/keepalived-ha-loadbalancer/

In my quest to make my services highly available I decided to use keepalived.keepalived is a framework for both load balancing and high availability that implements VRRP.This is a protocol that you see on some routers and has been implemented in keepalived. It creates a Virtual IP (or VIP, or floating IP) that acts as a gateway to route traffic to all participating hosts.This VIP that can provide a high availability setup and fail over to another host in the event that one is down. In this video, we’ll set up and configure keepalived, we’ll test our configuration to make sure it’s working, and we’ll also talk about some advanced use cases like load balancing.

https://technotim.live/posts/keepalived-ha-loadbalancer/

**Comment by layer3lv** [comment]
missing 
/etc/sysctl.conf
net.ipv4.ip_nonlocal_bind=1
more info: https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/load_balancer_administration/s1-initial-setup-forwarding-vsa

**Comment by morganmf** [comment]
Hello Tim
 I've followed ur instructions and all works fine although I would like to balance the load between pihole instead of just using one and leting the other idle. Any hints ? 
I ve got three piholes, two on proxmox and one on HP thinclient running ubuntu server. One proxmox pihole is tansy-1, tc pihole is tansy-2, the other proxmox is tansy-4, first two are HA as tansy-5.. I wish I could spread the load between tansy-1 and tansy-2, in case one of them is down the other one will take full load.
